Source/Core/VideoBackends/OGL/OGLTexture.h

```diff
@@ -0,0 +1,37 @@
+#pragma once
+
+#include "Common/GL/GLUtil.h"
+
+#include "VideoCommon/AbstractTexture.h"
+#include "VideoCommon/BPStructs.h"
+#include "VideoCommon/VideoCommon.h"
+
+namespace OGL
+{
+
+class OGLTexture : public AbstractTexture
+{
+public:
+
+  GLuint texId;
+  GLuint framebuffer;
+
+  OGLTexture(const TextureConfig& config);
+  ~OGLTexture();
+
+  virtual void Bind(unsigned int stage) override;
+  virtual bool Save(const std::string& filename, unsigned int level) override;
+
+  virtual void CopyRectangleFromTexture(const AbstractTexture* source,
+    const MathUtil::Rectangle<int>& srcrect,
+    const MathUtil::Rectangle<int>& dstrect) override;
+  virtual void Load(const u8* buffer, u32 width, u32 height, u32 expanded_width,
+    u32 level) override;
+  virtual void FromRenderTarget(bool is_depth_copy, const EFBRectangle& srcRect, bool scaleByHalf,
+    unsigned int cbufid, const float* colmat) override;
+
+  static void DisableStage(unsigned int stage);
```

Comment:
These are called by outside classes (TextureConverter, Render, etc).  It will probably need to be cleaned up later.  A large majority of this code was following the design originally dictated by phire.  I modified it where I felt like I could.